### PR TITLE
lib-classifier: Refactor survey task Choice and Questions styling

### LINKED ISSUE #6239: Survey task: Choice and Questions styling improvements
## Package
_Choose from the list:_
- [ ] app-content-pages
- [ ] app-project
- [ x ] lib-classifier
- [ ] lib-panoptes-js
- [ ] lib-react-components
- [ ] unknown

## Is your feature request related to a problem? Please describe.
This is in response to feedback stating it takes too much scrolling to see all the choices. 

Figma https://www.figma.com/design/ASqNlLhicNRzEDhYDA2t8j/Survey?node-id=0-1&m=dev&t=PxDxXDB67AH8V9Aa-1 

## Describe the solution you'd like
Increasing the width to match the task area at 540px helps a lot. 
Standardizing the width of buttons and spacing gives a neater presentation. 
Allowing for the animal description and 'Often Confused with' to be collapsed allows power users a shortcut past unnecessary info.
All fonts have been increased to match our new standard of 1rem / 16pt. The individual animal name is 24pt, bold. 

Default view w/ collapsed animal description and 'Often Confused with'
Screenshot 2024-08-27 at 4 10 51 PM

Screenshot 2024-08-27 at 3 47 50 PM

Each button has been standardized to 110px W, 40px H. When the copy is longer than that width will allow, please increase the widths by 10px increments (120, 130, 140, etc) with no less than 20px padding.

Screenshot 2024-08-27 at 3 56 01 PM Screenshot 2024-08-27 at 3 54 48 PM

Before and after:
Screenshot 2024-08-27 at 4 10 35 PM

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
- closes #6239 
- related to #6238 and #6076 - whose related PR #6494 should be merged to production together with this PR

## Describe your changes
- refactors Choice and related components per related Figma: https://www.figma.com/design/ASqNlLhicNRzEDhYDA2t8j/Survey?node-id=55-7053&t=BvyK5fjoLo5qzkDS-4

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - lib-classifier, staging: https://local.zooniverse.org:8080/?project=1634&workflow=2434
- What user actions should my reviewer step through to review this PR?
  - compare Figma designs to local survey task
  - note new collapsable info (choice description and confused withs)
- Which storybook stories should be reviewed?
  - http://localhost:6006/?path=/story/tasks-survey-choice--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see related survey task feedback follow-up issues

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

## Post-merge
- [ ] This PR adds translations keys to English dictionary(s). See guidance for pulling new keys to Lokalise [here](https://github.com/zooniverse/how-to-zooniverse/blob/master/Translations/lokalise.md#lokalise-and-fem).

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js [hunk 1]
```diff
@@ -46,6 +58,24 @@ function Choice({
...
+          <FormUp />
+        </>
+      ) : (
+        <>
+          <Text>{t('SurveyTask.Choice.moreInfo')}</Text>
+          <FormDown />
+        </>
+      )}
+    </Box>
+  )
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:78]: The text and icon alignment isn't quite right, needs a little adjustment.

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js [hunk 2]
```diff
@@ -85,57 +114,92 @@ function Choice({
...
+          gap='xsmall'
+          justify='center'
+          margin={{ top: '30px' }}
+          pad={{ top: 'small' }}
+        >
+          <StyledButton
+            fill='horizontal'
+            label={t('SurveyTask.Choice.cancel')}
+            onClick={() => handleDelete(choiceId)}
+          />
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:194]: Previously, if a previously selected choice was selected again, then "Not this" deleted that choice selection. Should "Cancel" still have the same delete choice selection action for previously selected choices?

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js]: Fixed!

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js]: Resolved in [090eeb9](https://github.com/zooniverse/front-end-monorepo/pull/6478/commits/090eeb95d121c4ffdff3f938c483760e69834e0b).

**Comment by seanmiller26** [comment]
> There is an extra 10px margin below the last line of Choices. Can we remove that OR reduce the margin above the next heading to be 10px? (Trying to tighten up the overall height of the component wherever possible, to fit entirely on the screen)

This  is still unresolved. Let's reduce the gap between choice sections to 20px instead of 30.
 Screenshot 2024-12-11 at 12 14 29 PM Screenshot 2024-12-11 at 12 14 17 PM  


I noticed choices with a longer name are not pill shaped at mobile size.

Desktop vs Mobile
Screenshot 2024-12-11 at 12 15 40 PM  Screenshot 2024-12-11 at 12 14 53 PM

'Confused with' choices do not have the correct margins and padding (tested with Woodpecker Cavity Cam)
Screenshot 2024-12-11 at 12 06 09 PM

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.stories.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ export const Default = ({ choiceId }) => {
           light: 'neutral-6'
         }}
         pad='1em'
-        width='380px'
+        width='540px'
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.stories.js:33]: Should be:
```
margin='20px'
width='498px'
```

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js [hunk 4]
```diff
@@ -85,57 +119,91 @@ function Choice({
...
+              light: 'neutral-7'
+            }}
+            margin='none'
+            size='1.5rem'
+            weight='bold'
+          >
+            {strings.get(`choices.${choiceId}.label`)}
+          </Heading>
+          <Button
+            a11yTitle={showInfo ? t('SurveyTask.Choice.lessInfo') : t('SurveyTask.Choice.moreInfo')}
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:148]: Does this Button need an aria-label if it contains a span with text? Typically a button with only an svg icon would definitely need an aria-label, but is it needed here? (I don't have a definitive answer, just checking reasoning behind this choice to include aria)

**Review by goplayoutside3 — APPROVED** [review]
Looks good! I reviewed by looking at the survey task in Storybook. I have two non-blocking comments:

1. The aria-label question about the More Info button.
2. Is it possible to give the Carousel a min-height of ~160px? As long as there are choice images, the rest of the Choice component always renders before the carousel populates, resulting in a layout jump when the images finally appear.

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js (hunk 4 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:148]: I think you're correct, it does not need an aria-label with the span text, changed accordingly 👍

**Comment by mcbouslog** [comment]
> 2. Is it possible to give the Carousel a min-height of ~160px? As long as there are choice images, the rest of the Choice component always renders before the carousel populates, resulting in a layout jump when the images finally appear.

Yes I think that improves the overall Choice loading; change made ✅

**Pull request merged**